src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -28,111 +62,930 @@ const ActorInit En_Lift_Nuts_InitVars = {
+            CURRENT_DAY == 3) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 13);
+            Message_StartTextbox(play, 0x27F5, &this->actor);
+            this->textId = 0x27F5;
+        } else if (func_80AE9B8C() > 0) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 1);
+            Message_StartTextbox(play, 0x27F0, &this->actor);
+            this->textId = 0x27F0;
+        } else {
+            Message_StartTextbox(play, 0x27F1, &this->actor);
+            this->textId = 0x27F1;
+        }
+        this->actor.flags &= ~ACTOR_FLAG_10000;
+        func_80AEABF0(this);
+        switch (CURRENT_DAY) {
+            case 1:
+                gSaveContext.save.weekEventReg[14] |= 0x10;
+                break;
+            case 2:
+                gSaveContext.save.weekEventReg[14] |= 0x20;
+                break;
+            case 3:
+                gSaveContext.save.weekEventReg[14] |= 0x40;
+                break;
+            default:
+                break;
+        }
+    } else {
+        func_800B85E0(&this->actor, play, 200.0f, -1);
+    }
+}
+
+void func_80AEB828(EnLiftNuts* this) {
+    if (!func_80AE9B4C(0, 4)) {
+        func_80AE9B4C(1, 4);
+        this->unk_356 = 1;
+    }
+    if (this->actionFunc == func_80AE9F70) {
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 15);
+    }
+    this->unk_354 = 0;
+    this->actionFunc = func_80AEB8A4;
+}
+
+void func_80AEB8A4(EnLiftNuts* this, PlayState* play) {
+    if (this->unk_354 == 22) {
+        if (this->unk_356 == 1) {
+            Message_StartTextbox(play, 0x27F6, &this->actor);
+            this->textId = 0x27F6;
+        }
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 17);
+        this->actionFunc = func_80AEB934;
+    }
+    this->unk_354++;
+}
+
+void func_80AEB934(EnLiftNuts* this, PlayState* play) {
+    if (Animation_OnFrame(&this->skelAnime, 5.0f)) {
+        func_80AEB9E0(this, play);
+    }
+}
+
+void func_80AEB974(EnLiftNuts* this) {
+    s16 textId = this->textId;
+
+    switch (textId) {
+        default:
+            this->unk_1E4 = 0;
+            break;
+        case 0x27EE:
+        case 0x27EF:
+            this->unk_1E4 = 1;
+            break;
+        case 0x27F5:
+            this->unk_1E4 = 2;
+            break;
+        case 0x27F4:
+            if (this->unk_354 == 0) {
+                this->unk_1E4 = 2;
+                break;
+            }
+            this->unk_1E4 = 1;
+    }
+}
+
+void func_80AEB9E0(EnLiftNuts* this, PlayState* play) {
+    static Color_RGBA8 sPrimColor = { 170, 130, 90, 255 };
+    static Color_RGBA8 sEnvColor = { 100, 60, 20, 255 };
+
+    Vec3f worldPos;
+    Vec3f velocity;
+    Vec3f accel;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+    s32 i;
+
+    worldPos = this->actor.world.pos;
+
+    primColor = sPrimColor;
+    envColor = sEnvColor;
```

Comment:
```suggestion
void func_80AEB9E0(EnLiftNuts* this, PlayState* play) {
    Color_RGBA8 primColor = { 170, 130, 90, 255 };
    Color_RGBA8 rnvColor = { 100, 60, 20, 255 };

    Vec3f worldPos;
    Vec3f velocity;
    Vec3f accel;
    s32 i;

    worldPos = this->actor.world.pos;
```
Would this match? I remember there being something meaningful if you just have an array that's immediately assigned to a temp. It's been a while since I've seen it though

Reply:
Small note, you might have to put `Vec3f worldPos = this->actor.world.pos;` above the colors for it to work.

Reply:
i thought this might work too but it doesnt seem to